src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,67 @@
+using Flux
+# I haven't checked ``downsample`` parameter as it is always performed after every block.
+function resnet()
+  layers = Chain(
+      Conv((7, 7), 3=>64, stride=(2, 2), pad=(3, 3)),
+      BatchNorm(64, λ=relu),
+      MaxPool((3, 3), stride=(2, 2), pad=(1, 1))
+      )
+  expansion = 1;
+  if groups != 1 || base_width != 64
+    error('Basic block only supports groups=1 and base_width=64');
+  end
+  if dilation > 1
+    error("Dilation > 1 not supported in BasicBlock");
+  end
+  planes = 64;    
+  layers = SkipConnection(layers, Chain(
+    Conv((3, 3), planes ÷ 2 => planes, stride = (1,1), pad = (1,1), dilation = 1),
+    BatchNorm(planes, λ = relu),
+    Conv((3, 3), planes => planes, stride = (1,1), pad = (1,1), dilation = 1),
+    BatchNorm(planes),
+    Sequential(Conv((1, 1), planes ÷ 2 => planes * expansion, stride=(1, 1)),
+           BatchNorm(planes * expansion)
+    ), x -> relu.(x)
```

Comment:
I think you mean `Chain` not `Sequential`. Also, these 1x1 convolutions only exist in certain ResNet models.

Reply:
No, its the downsampling layer it should use ``Sequential``. Infact both the basic and bottleneck blocks should wrap the layers in ``Sequential``. I have changed that.

>  Also, these 1x1 convolutions only exist in certain ResNet models, so that's something to keep in mind for the configs.

Ok, I will check this.
I hope the Adaptive pooling PR gets merged so that I can test these on my system.

Reply:
What is `Sequential`?

Reply:
oh, My bad! I got confused between PyTorch's ``sequential`` and Flux's ``Chain`` functions and assumed that Flux has got ``sequential`` function and PyTorch has got otherwise.

Reply:
No problem, that's what I figured. `Chain` in Flux <--> `torch.Sequential`.